### ISSUE TITLE
Multiple steps in if_ WorkChain block will lead to corrupted stepper in pickle

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#902]
When pickling a `WorkChain` which contains an `if_` block with more than one line in the block body, the stepper will be written incorrectly, leading to the pickle being corrupt. When trying to recreate the `Process` with `create_from()` the method will fail.
A minimal example to demonstrate this problem is the following:

```
class ParentWorkChain(WorkChain):

    @classmethod
    def define(cls, spec):
        super(ParentWorkChain, cls).define(spec)
        spec.outline(
            cls.step,
            cls.resulta
        )

    def step(self):
        running = submit(SubWorkChain)
        self.report('Submitted {}<{}>'.format(SubWorkChain.__name__, running.pid))
        return ToContext(sub=running)

    def resulta(self):
        self.report('ParentWorkChain finished')

class SubWorkChain(WorkChain):

    @classmethod
    def define(cls, spec):
        super(SubWorkChain, cls).define(spec)
        spec.outline(
            if_(cls.conditional)(
                cls.do_print,
                cls.do_print,
            ),
            cls.step
        )

    def conditional(self):
        return True

    def do_print(self):
        self.report('testing')

    def step(self):
        self.report('BaseWorkChain running')

run(ParentWorkChain)
```

The `ParentWorkChain` will `submit` the `SubWorkChain` causing it to be pickled. Since the `SubWorkChain` has two steps in the block body of the conditional, the saved state in the pickle will contain a corrupted or incomplete stepper. If you remove one of the two `cls.do_print` calls from the `outline` of the `SubWorkChain` everything works expected. The same goes for simply running the `SubWorkChain` directly in which case of course there is no pickling going on.

**Comment by sphuber** [comment]
I loaded up one of the corrupted pickles produced by the example workchains above and printed the `stepper_state` which prints the following:

```
with open('2348.pickle', 'r') as handle:
    cp = pickle.load(handle)
print cp['stepper_state']

<Bundle {'stepper_pos': <Bundle {'stepper_pos': <Bundle {'pos': 1}>, 'pos': 1}>, 'pos': 3}>
```

I think the problem is that the logic for saving the stepper state of `_If` instructions or the loading thereof is incomplete or broken.

**Comment by sphuber** [comment]
Fixed in #904
